metabase/metabase
# Fix boolean parameter parsing from query string in dashboards

Fixes https://github.com/metabase/metabase/issues/45670

Extracts functions from selectors that depend on the dashboard being already loaded and use them to parse query string parameters according to the mapped fields.

How to verify:
- You need a db schema with boolean fields. Postgres QA DB can be used for that.
- Create a new SQL query `SELECT id, boolean FROM many_data_types WHERE {{boolean}}` and map `boolean` to a boolean field in the DB -> Save as Q1
- Create a dashboard, add Q1, add a "Text or Category" filter and map it to the boolean field -> Save
- Make sure that you can pass boolean values via query string

<img width="916" alt="Screenshot 2024-10-23 at 16 00 32" src="https://github.com/user-attachments/assets/f924a8a2-5a56-4d73-bcae-ccaefc22d745">
<img width="835" alt="Screenshot 2024-10-23 at 16 00 42" src="https://github.com/user-attachments/assets/bd4f0a64-ece2-4991-be36-2a7d94d6bd5e">
